### PR TITLE
fix(condition-tree): longer than and shorter than operators are supported

### PR DESCRIPTION
**Pull request by Scra3**
…rted

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [ ] Ensure that Types have been updated according to your changes (if needed)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-toolkit/src/validation/condition-tree.ts [hunk 1]
```diff
@@ -114,10 +114,13 @@ export default class ConditionTreeValidator {
...
     columnSchema: ColumnSchema,
   ): void {
-    const { value, field } = conditionTree;
+    const { value, field, operator } = conditionTree;
     const { columnType } = columnSchema;
     const allowedTypes = MAP_ALLOWED_TYPES_FOR_COLUMN_TYPE[columnType as PrimitiveTypes];
 
-    FieldValidator.validateValue(field, columnSchema, value, allowedTypes);
+    // exclude some cases where the value is not related to the columnType of the field
+    if (operator !== 'ShorterThan' && operator !== 'LongerThan') {
```
Comment (romain-gilliotte) [inline packages/datasource-toolkit/src/validation/condition-tree.ts:122]: That's also the case for other operators

BeforeXDaysAgo is on date, but the value the operator expects is an integer

Comment (Scra3) [reply packages/datasource-toolkit/src/validation/condition-tree.ts]: 👍

File: packages/datasource-toolkit/src/validation/rules.ts [hunk 2]
```diff
@@ -61,8 +68,8 @@ export const MAP_ALLOWED_TYPES_FOR_COLUMN_TYPE: Readonly<
   Number: ['Number', ValidationTypesArray.Number, ValidationPrimaryTypes.Null],
   Boolean: ['Boolean', ValidationTypesArray.Boolean, ValidationPrimaryTypes.Null],
   Enum: ['Enum', ValidationTypesArray.Enum, ValidationPrimaryTypes.Null],
-  Date: ['Date', 'Number', ValidationPrimaryTypes.Null],
-  Dateonly: ['Dateonly', 'Number', ValidationPrimaryTypes.Null],
+  Date: ['Date', ValidationPrimaryTypes.Null],
```
Comment (Scra3) [inline packages/datasource-toolkit/src/validation/rules.ts:71]: Not really sure but Number should not be accepted.

Comment (romain-gilliotte) [reply packages/datasource-toolkit/src/validation/rules.ts:71]: previous bug fix for AfterXHoursAgo

File: packages/datasource-toolkit/src/validation/rules.ts [hunk 3]
```diff
@@ -43,7 +43,14 @@ export const MAP_ALLOWED_OPERATORS_FOR_COLUMN_TYPE: Readonly<
...
   ],
-  Number: [...BASE_OPERATORS, ...ARRAY_OPERATORS, 'GreaterThan', 'LessThan'],
+  Number: [
+    ...BASE_OPERATORS,
+    ...ARRAY_OPERATORS,
+    'GreaterThan',
+    'LessThan',
+    'ShorterThan',
+    'LongerThan',
+  ],
```
Comment (romain-gilliotte) [inline packages/datasource-toolkit/src/validation/rules.ts:53]: you can remove shorterthan and longerthan

**Review by romain-gilliotte — CHANGES_REQUESTED** [review]
Now that the validator is fixed, can you call it at the following line?

https://github.com/ForestAdmin/agent-nodejs/blob/beta/packages/datasource-toolkit/src/decorators/validation/collection.ts#L62

**Review by romain-gilliotte — APPROVED** [review]

**Pull request merged**